### DISCUSSION TITLE
キーボードとtextfieldの重なりについて

### DISCUSSION THREAD

**Discussion by atsujr** [lit-iphone-course/lit-iphone-course#49] [Question-original] [answered]
(1) 実現したいこと
キーボードが出てきた時に、キーボードでtextfieldが隠れないようにしたい
(2) 実現するために自分が試した内容とその結果
以下のURLを参考に、実装しようとしてみたが、実装してもviewがずれなかった
(3) 調べた内容や資料
https://qiita.com/junshu310/items/6a8bac1ec598715c16c1
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
    self.view.transform = CGAffineTransform(translationX: 0, y: keyboardTopPositionY - textFieldTopPositionY)
この部分で、
keyboardTopPositionY と textFieldTopPositionYが宣言されてないというエラーが出るため、サイトにてどこで宣言しているか確認したが、どこにも宣言されておらず、Topを消して実装してみたところ、エラーは消えたので実行してみたが、うまくviewがずれなかったです。
ご確認よろしくお願いします。githubのURLは、以下の通りです。
https://github.com/atsujr/BookList

**Comment by yukiny0811** [comment]
サイトの中の2章のところに書いてあるよー！
<img width="1044" alt="image" src="https://user-images.githubusercontent.com/28947703/192096029-a005d82c-5b8d-4df0-9e95-92a9497c331c.png">


```.swift
guard let textField = _activeTextField else { return }
// キーボード、画面全体、textFieldのsizeを取得
let rect = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue
guard let keyboardHeight = rect?.size.height else { return }
let mainBoundsSize = UIScreen.main.bounds.size
let textFieldHeight = textField.frame.height

// ①
let textFieldPositionY = textField.frame.origin.y + textFieldHeight + 10.0
// ②
let keyboardPositionY = mainBoundsSize.height - keyboardHeight
```

**Reply by atsujr** [reply]
なるほど、そしたらこの部分って、textFieldPositionYとtextFieldTopPositionYは同じで、keyboardTopPositionYとkeyboardPositionYは同じように書かないといけないという認識で間違えてないでしょうか？
この場合、textFieldPositionYとtextFieldTopPositionY、keyboardTopPositionYとkeyboardPositionYはどっちかに統一しないといけないということでしょうか？

**Reply by yukiny0811** [reply]
あ！ごめんね！Topがついてるのとついてないのがあるのか！
どっちかに統一すれば動くと思ってたけど、、動かない？

**Comment by txxxxc** [comment, marked as answer]
あ、この問題なんだけど前JrとDiscordで話したついでにサンプルコード渡したから一旦実装の目処は立ってそう！


@atsujr 
解決したタイミングでもう一回教えてもらえると嬉しい！
https://github.com/Tomoya113/KeyboardAutoScroll
